pkg/kv/kvserver/below_raft_protos_test.go

```diff
@@ -14,53 +14,32 @@ import (
 	"fmt"
 	"hash/fnv"
 	"math/rand"
-	"reflect"
+	"regexp"
 	"testing"
 
 	"github.com/cockroachdb/cockroach/pkg/roachpb"
 	"github.com/cockroachdb/cockroach/pkg/storage/enginepb"
+	"github.com/cockroachdb/cockroach/pkg/testutils"
+	"github.com/cockroachdb/cockroach/pkg/testutils/echotest"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
 	"github.com/cockroachdb/cockroach/pkg/util/protoutil"
+	"github.com/stretchr/testify/require"
 	"go.etcd.io/etcd/raft/v3/raftpb"
 )
 
-func verifyHash(b []byte, expectedSum uint64) error {
-	hash := fnv.New64a()
-	if _, err := hash.Write(b); err != nil {
-		return err
-	}
-	if sum := hash.Sum64(); sum != expectedSum {
-		return fmt.Errorf("expected sum %d; got %d", expectedSum, sum)
-	}
-	return nil
-}
-
-// An arbitrary number chosen to seed the PRNGs used to populate the tested
-// protos.
-const goldenSeed = 1337
-
-// The count of randomly populated protos that will be concatenated and hashed
-// per proto type. Given that the population functions have a chance of leaving
-// some fields zero-valued, this number must be greater than `1` to give this
-// test a reasonable chance of encountering a non-zero value of every field.
-const itersPerProto = 20
-
-type fixture struct {
-	populatedConstructor   func(*rand.Rand) protoutil.Message
-	emptySum, populatedSum uint64
-}
+// TestBelowRaftProtosDontChange is a manually curated list of protos that we
+// use below Raft. Care must be taken to change these protos, as replica
+// divergence could ensue (if the old code and the new code handle the updated
+// or old proto differently). Changes to the encoding of these protos will be
+// detected by this test. The expectations should only be updated after a
+// reflection on the safety of the proposed change.
+func TestBelowRaftProtosDontChange(t *testing.T) {
```

Comment:
Can we make this list more visible somewhere, e.g. where the Protobufs are defined or something? Or at least add a comment reminding readers to update this list?

Reply:
Not sure, as you can see from the test the protos are defined in `enginepb`, `roachpb`, `kvserverpb`. I get the discoverability problem but I don't see anything obvious that I can just add to this PR. It seems overkill to create per-package registries, linters, etc, at this point.